ext/rugged/rugged_commit.c

```diff
@@ -612,11 +612,53 @@ static VALUE rb_git_commit_header(VALUE self)
 	return rb_str_new_utf8(raw_header);
 }
 
+/*
+ *  call-seq:
+ *    Rugged::Commit.extract_signature(repo, commit, field_name) -> [str, str]
+ *
+ *  Returns +commit+'s signature in 'field' and the signed data
+ *
+ *  The signature is done over the contents of the commit without the
+ *  signature block in the header, which is the data in the second
+ *  element in the return array.
+ */
+static VALUE rb_git_commit_extract_signature(VALUE self, VALUE rb_repo, VALUE rb_commit, VALUE rb_field)
+{
+	int error;
+	VALUE ret_arr;
+	git_oid commit_id;
+	git_repository *repo;
+	git_buf signature = {0}, signed_data = {0};
+
+	rugged_check_repo(rb_repo);
+	Data_Get_Struct(rb_repo, git_repository, repo);
+
+	error = git_oid_fromstr(&commit_id, StringValueCStr(rb_commit));
+	rugged_exception_check(error);
+
+	error = git_commit_extract_signature(&signature, &signed_data, repo, &commit_id, StringValueCStr(rb_field));
+	if (error < 0) {
+		git_buf_free(&signature);
+		git_buf_free(&signed_data);
+	}
+
+	rugged_exception_check(error);
+
+	ret_arr = rb_ary_new3(2, rb_enc_str_new(signature.ptr, signature.size, rb_utf8_encoding()),
```

Comment:
actually, since we're meant to verify these bytes, they'd be better off as a binary/ascii string instead.